connectors/services/job_scheduling.py

```diff
@@ -171,48 +173,67 @@ async def _should_schedule_on_demand_sync():
 
     async def _scheduled_sync(self, connector):
         @with_concurrency_control()
-        async def _should_schedule_scheduled_sync():
+        async def _should_schedule_scheduled_sync(job_type):
             try:
                 await connector.reload()
             except DocumentNotFoundError:
                 logger.error(f"Couldn't reload connector {connector.id}")
                 return False
 
-            now = datetime.utcnow()
             if (
-                connector.last_sync_scheduled_at is not None
-                and connector.last_sync_scheduled_at > now
+                job_type == JobType.ACCESS_CONTROL
```

Comment:
For incremental sync, I will check both the feature flag and this value:
https://github.com/elastic/connectors-python/blob/e4ac7dc0f6b4b674eefc1787868524c48a8085af/connectors/source.py#L342

Not sure whether this applies to DLS.

What do you think we move this check to line 99:
https://github.com/elastic/connectors-python/blob/e4ac7dc0f6b4b674eefc1787868524c48a8085af/connectors/services/job_scheduling.py#L94-L101

This way, we can 1) access `source_klass. support_incremental_sync` directly, 2) avoid `if...else...` in `_should_schedule_scheduled_sync`.

Reply:
Makes sense, adjusted it and did some other simplifications 👍 

> Not sure whether this applies to DLS.

No, this doesn't apply to DLS. We can still discuss it, but this should be a separate change from this PR.